runtime/compiler/p/codegen/J9PPCWatchedStaticFieldSnippet.cpp

```diff
@@ -0,0 +1,127 @@
+      {
+      cg()->addExternalRelocation(
+         new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor + offsetof(J9JITWatchedStaticFieldData, method), reinterpret_cast<uint8_t *>(staticFieldData.method), reinterpret_cast<uint8_t *>(TR::SymbolType::typeMethod), TR_SymbolFromManager, cg()),
+         __FILE__,
+         __LINE__,
+         node);
+      }
+   else
+      {
+      cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor + offsetof(J9JITWatchedStaticFieldData, method), NULL, TR_RamMethod, cg()), __FILE__, __LINE__, node);
+      }
+
+   bool isResolved = !node->getSymbolReference()->isUnresolved();
+   // If the field is unresolved then we populate these snippet fields with the correct value at runtime (via the instructions generated by generateFillInDataBlockSequenceForUnresolvedField)
+   // and hence don't need to add relocation records here.
+   if (isResolved)
+      {
+      cg()->addExternalRelocation(
+         new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor + offsetof(J9JITWatchedStaticFieldData, fieldAddress), reinterpret_cast<uint8_t *>(node->getSymbolReference()), reinterpret_cast<uint8_t *>(node->getInlinedSiteIndex()), TR_DataAddress, cg()),
+         __FILE__,
+         __LINE__,
+         node);
+      if (cg()->comp()->getOption(TR_UseSymbolValidationManager))
+         {
+         cg()->addExternalRelocation(
+               new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor + offsetof(J9JITWatchedStaticFieldData, fieldClass), reinterpret_cast<uint8_t *>(staticFieldData.fieldClass), reinterpret_cast<uint8_t *>(TR::SymbolType::typeClass), TR_SymbolFromManager, cg()),
+               __FILE__,
+               __LINE__,
+               node);
+         }
+      else
+         {
+         /* Deploy when AOT is Enabled. TR_ClassAddress needs to support contiguous addresses. 
+         TR_RelocationRecordInformation *recordInfo = (TR_RelocationRecordInformation *)cg()->comp()->trMemory()->allocateMemory(sizeof(TR_RelocationRecordInformation), heapAlloc);
+         recordInfo->data1 = (uintptr_t)(node->getSymbolReference());
+         recordInfo->data2 = (uintptr_t)(node->getInlinedSiteIndex());
+         recordInfo->data3 = (uintptr_t)(fixedSequence7);
+         cg()->addExternalRelocation(
+               new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor + offsetof(J9JITWatchedStaticFieldData, fieldClass), (uint8_t *)recordInfo, TR_ClassAddress, cg()),
+               __FILE__,
+               __LINE__,
+               node);
+         */
+         }
+      }
+
+      // Insert into TOC 
+      if (TR::Compiler->target.is64Bit() && getTOCOffset() != PTOC_FULL_INDEX)
+         {
+         TR_PPCTableOfConstants::setTOCSlot(getTOCOffset(), reinterpret_cast<uintptrj_t>(cursor));
+         }
+      else if (getLowerInstruction() != NULL)
+         {
+         // Handle Nibles - Generation of instructions to materialise address
```

Comment:
Thanks for finding the typo. Done - please confirm.

Reply:
@AlenBadel it can be helpful to reviewers if you specify which commit the changes were made in within a comment. Otherwise we have to dig around and find it ourselves, which can be time consuming. GitHub will hyperlink a SHA automatically if you paste at least 7 characters of it. For example:

Fixed in 11604f7be979649b65749f4acc22064650b9b3f5